src/connection.rs

```diff
@@ -185,27 +240,28 @@ impl<T: AsyncRead + AsyncWrite + Unpin> Receiver<T> {
     /// values. If PONGs are not expected or uninteresting,
     /// [`Receiver::receive_data`] may be used instead which skips over PONGs
     /// and considers only application payload data.
-    pub async fn receive(&mut self, message: &mut Vec<u8>) -> Result<Incoming<'_>, Error> {
+    pub async fn receive(&mut self, token: RecvToken, message: &mut Vec<u8>) -> Result<(Incoming<'_>, RecvToken), Error> {
```

Comment:
```suggestion
    pub async fn receive(&mut self, token: RecvToken, message: &mut Vec<u8>) -> Result<Received<'_>, Error> {
```
with
```rust
pub enum Received<'a> {
    /// Text or binary data.
    Data(Data, RecvToken),
    /// Data sent with a PONG control frame.
    Pong(&'a [u8], RecvToken)
}

impl<'a> Received<'a> {
    pub fn ignore(self) -> RecvToken {
        match self {
            Received::Data(_, t) => t,
            Received::Pong(_, t) => t,
    }
    ....
}
```
i.e. `Received` would replace `Incoming` with the token in the constructors. Just a stylistic idea, not sure whether it is preferable.

Reply:
I was thinking about that initially, but decided against it as I felt that the token should be visible in the API.